tests/v2/validation/projects/projects.go

```diff
@@ -0,0 +1,184 @@
+	prtb.ProjectName = projectName
+	prtb.RoleTemplateName = role
+	prtb.UserPrincipalName = user.PrincipalIDs[0]
+	createdProjectRoleTemplateBinding, err := projectsApi.CreateProjectRoleTemplateBinding(client, &prtb)
+	if err != nil {
+		return nil, err
+	}
+
+	return createdProjectRoleTemplateBinding, nil
+}
+
+func waitForFinalizerToUpdate(client *rancher.Client, projectName string) error {
+	err := kwait.Poll(defaults.FiveHundredMillisecondTimeout, defaults.FiveSecondTimeout, func() (done bool, pollErr error) {
+		project, pollErr := projectsApi.ListProjects(client, project.Namespace, metav1.ListOptions{
+			FieldSelector: "metadata.name=" + projectName,
+		})
+		if pollErr != nil {
+			return false, pollErr
+		}
+
+		if len(project.Items[0].Finalizers) == 2 {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkPodLogsForErrors(client *rancher.Client, cluster string, podName string, namespace string, errorPattern string, startTime time.Time, endTime time.Time) error {
+	startTimeUTC := startTime.UTC()
+	endTimeUTC := endTime.UTC()
+
+	podLogs, err := kubeconfig.GetPodLogs(client, cluster, podName, namespace)
+	if err != nil {
+		return err
+	}
+
+	delimiter := "\n"
+	segments := strings.Split(podLogs, delimiter)
+	errorRegex := regexp.MustCompile(errorPattern)
+	timeRegex := regexp.MustCompile(`^\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}`)
+
+	for _, segment := range segments {
+		timeMatches := timeRegex.FindStringSubmatch(segment)
+		if len(timeMatches) > 0 {
+			segmentTime, err := time.Parse(timeFormat, timeMatches[0])
+			if err != nil {
+				continue
+			}
+
+			segmentTimeUTC := segmentTime.UTC()
+
+			if segmentTimeUTC.After(startTimeUTC) && segmentTimeUTC.Before(endTimeUTC) {
+				if matches := errorRegex.FindStringSubmatch(segment); len(matches) > 0 {
+					return errors.New("error logs found in rancher: " + segment)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getRancherLeaderPod(client *rancher.Client) (string, error) {
+	configMapList, err := client.Steve.SteveType(configmaps.ConfigMapSteveType).NamespacedSteveClient(kubeSystemNamespace).List(nil)
+	if err != nil {
+		return "", err
+	}
+
+	var leaderAnnotation string
+	for _, cm := range configMapList.Data {
+		if cm.Name == rancherConfigMap {
+			leaderAnnotation = cm.Annotations[rancherLeaderAnnotation]
+			break
+		}
+	}
+
+	var leaderRecord map[string]interface{}
+	err = json.Unmarshal([]byte(leaderAnnotation), &leaderRecord)
+	if err != nil {
+		return "", err
+	}
+
+	leaderPodName := leaderRecord["holderIdentity"].(string)
+
+	return leaderPodName, nil
+}
+
+func updateProjectNamespaceFinalizer(client *rancher.Client, projectName string, namespace string, finalizer []string) (*v3.Project, error) {
+	var updateProject = v3.Project{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       projectName,
+			Namespace:  namespace,
+			Finalizers: finalizer,
+		},
+		Spec: v3.ProjectSpec{
+			ClusterName: namespace,
+		},
```

Comment:
Ca we not reuse the project var declared here? https://github.com/rancher/rancher/blob/6cd6a86e9a8c9edfffcf003a4ce252db3a029069/tests/v2/validation/projects/projects.go#L38 

We can send in the empty finalizers for the other functions while creating projects?